library/init/meta/lean_parser.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+-/
+prelude
+import init.category.monad init.category.alternative init.meta.pexpr
+
+meta constant lean_parser : Type → Type
+
```

Comment:
@leodemoura I think "base types" are fine. I was commenting more on the proliferation of internal APIs in the global namespace.